client: append to buffer list to save all result from wildcard command

jinmyeonglee opened the pull request:
https://tracker.ceph.com/issues/61819


<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

jinmyeonglee commented:
jenkins test make check arm64

vshankar commented on src/client/Client.cc:
```diff
@@ -6368,7 +6368,7 @@ void Client::handle_command_reply(const MConstRef<MCommandReply>& m)
 
   auto &op = command_table.get_command(tid);
   if (op.outbl) {
-    *op.outbl = m->get_data();
```
https://tracker.ceph.com/issues/61819#note-4

Good catch. Another question: What if one of the ranks return an error? How is that handled?

jinmyeonglee replied:
Venky Shankar wrote:
> Jinmyeong Lee wrote:
> > Jinmyeong Lee wrote:
> > > https://github.com/ceph/ceph/pull/52200
> > 
> > I also want to discuss how to append the result in bufferlist.
> > From now, I just append the result in the bufferlist, but I think the below format would be better. (Also afraid of side-effect)
> > 
> > {
> >     tid_1: [],
> >     tid_2: [],
> >     ...
> > }
> 
> Is the `tid_*` prefix the transaction-id? IMO, using mds rank or IDs would be better.

I agree. To make a similar result with a result of "ceph tell mds.* session ls" command.

// For all active MDSs:
{
    "mds.$id_0": [],
    "mds.$id_1": [],
...
}

I think if some ranks return error, then it is okay to skip to add to the outbl. -> I will add a commit for this.
When I send "mds.* session ls" command using ceph CLI, the standby MDSs return -22(EINVAL) and the result is not in CLI command.

But to avoid a side effects about a single target mds command, I think it would be good to add a new multi_target_id to common/CommandTable to confirm whether all the commands end. (so it can make possible to make the above format of result.)

So,
In common/CommandTable,
* Add new functions: ceph_tid_t get_new_multi_command_id(), bool is_all_multi_commands_end(ceph_tid_t multi_target_id)
* Adjust: T& start_command(ceph_tid_t multi_target_id=0)
* Add a few lines of code in Client::mds_command and Client::handle_command_reply
- mds_command
```
ceph_tid_t multi_target_id = 0;
  if (non_laggy.size() > 1) {
      std::scoped_lock cmd_lock(command_lock);
      multi_target_id = command_table.get_new_multi_command_id();
  }
...
```
- handle_command_reply: append the command result to bufferlist with format

jinmyeonglee replied:
@vshankar PING: Could you give feedback about this approach?

vshankar replied:
Sure. Sorry - was on PTO. Just back. Will have a look.

vshankar replied:
> Venky Shankar wrote:
> 
> > Jinmyeong Lee wrote:
> > > Jinmyeong Lee wrote:
> > > > #52200
> > > 
> > > 
> > > I also want to discuss how to append the result in bufferlist.
> > > From now, I just append the result in the bufferlist, but I think the below format would be better. (Also afraid of side-effect)
> > > {
> > > tid_1: [],
> > > tid_2: [],
> > > ...
> > > }
> > 
> > 
> > Is the `tid_*` prefix the transaction-id? IMO, using mds rank or IDs would be better.
> 
> I agree. To make a similar result with a result of "ceph tell mds.* session ls" ceph CLI command.
> 
> // For all active MDSs: { "mds.$id_0": [], "mds.$id_1": [], ... }

Looks good.

> 
> I think if some ranks return error, then it is okay to skip to add to the outbl. -> I will add a commit for this. When I send "mds.* session ls" command using ceph CLI, the standby MDSs return -22(EINVAL) and the result is not in CLI command.

Should we add the failed errno as part of the json? Which means for each mds reply the json will have an errno and the command output. If errno is not zero, then the command output would be empty. What do you think?

> 
> But to avoid side effects about a single target mds command, I think it would be good to add a new multi_target_id to common/CommandTable to confirm whether all the commands end. (so it can make it possible to make the above format of the result.)

You mean the single target mds command now using the changed format, right?

jinmyeonglee replied:
> You mean the single target mds command now using the changed format, right?
I mean, No.
If we edit the result format of a single MDS target command, we should consider other parts using this Client like here(https://github.com/ceph/ceph/blob/main/src/ceph.in#L1252C20-L1254). I am a little afraid of these side effects.
Also, the result of "ceph tell mds.0 session ls" Ceph CLI command is only a list of sessions. (There is no key with "mds.0".) I think following the Ceph CLI results would be better.
So, I want to make a new format when just the mds_spec is "*", which means multi-target MDSs.


> Should we add the failed errno as part of the json? Which means for each mds reply the json will have an errno and the command output. If errno is not zero, then the command output would be empty. What do you think?

I briefly considered if it is possible, but in my opinion, it looks difficult.
In libcephfs, when "ceph_mds_command" function calls the "Client::mds_command" with mds_spec = "*", all command Messages are on one C_Safer_Cond.

And in "Client::handle_command_reply" function, the individual response from each MDS is handled and the errno is updated in "complete(r)". So in this situation, the libcephfs can get only one errorno from "cond.wait()".
(
You can check it here:
https://github.com/ceph/ceph/blob/main/src/libcephfs.cc#L543-L555
https://github.com/ceph/ceph/blob/main/src/client/Client.cc#L6334-L6347
https://github.com/ceph/ceph/blob/main/src/client/Client.cc#L6389-L6391
)

If we should keep all the errornos, we should introduce a new field to keep each errono in somewhere, like in common/CommandTable. Or introducing somewhat new Structure for multi-target commands? 😀

vshankar replied:
> Thanks for your feedback!!
> 
> > You mean the single target mds command now using the changed format, right?
> 
> I mean, No. If we edit the result format of a single MDS target command, we should consider other parts using this Client like here(https://github.com/ceph/ceph/blob/main/src/ceph.in#L1252C20-L1254). I am a little afraid of these side effects. Also, the result of "ceph tell mds.0 session ls" Ceph CLI command is only a list of sessions. (There is no key with "mds.0".) I think following the Ceph CLI results would be better. So, I want to make a new format when just the mds_spec is "*", which means multi-target MDSs.

Right - makes sense. I'm fine with your proposal of adding multi_target_it.

> 
> > Should we add the failed errno as part of the json? Which means for each mds reply the json will have an errno and the command output. If errno is not zero, then the command output would be empty. What do you think?
> 
> I briefly considered if it is possible, but in my opinion, it looks difficult. In libcephfs, when "ceph_mds_command" function calls the "Client::mds_command" with mds_spec = "*", all command Messages are on one C_Safer_Cond.

Ah, that's a bummer.

> 
> And in "Client::handle_command_reply" function, the individual response from each MDS is handled and the errno is updated in "complete(r)". So in this situation, the libcephfs can get only one errorno from "r = cond.wait()". ( You can check it here: https://github.com/ceph/ceph/blob/main/src/libcephfs.cc#L543-L555 https://github.com/ceph/ceph/blob/main/src/client/Client.cc#L6334-L6347 https://github.com/ceph/ceph/blob/main/src/client/Client.cc#L6389-L6391 )
> 
> If we should keep all the errornos, we should introduce a new field to keep each errono in somewhere, like in 
common/CommandTable. Or introducing somewhat new Structure for multi-target commands? 😀

Anyway, let's stick to your original idea.

jinmyeonglee replied:
Thanks for checking! I will make commits for this ASAP.

jinmyeonglee commented:
jenkins test make check

jinmyeonglee commented:
jenkins test windows

jinmyeonglee commented:
@vshankar I added commits, Could you please take a look?
(well, make check is failing for other reasons.)

vshankar commented:
> @vshankar I added commits, Could you please take a look? (well, make check is failing for other reasons.)

Sure. On it today.

vshankar commented:
@jinmyeonglee Did you consider adding some tests for this? I think it would be good to have some.

jinmyeonglee commented:
> @jinmyeonglee Did you consider adding some tests for this? I think it would be good to have some.

Good, I can. Do you mean under this directory? (https://github.com/ceph/ceph/tree/main/src/test/client)
And also, could you share a good example? (I think I cannot find original tests for sending commands thru libcephfs -> Client.)

vshankar commented:
> > @jinmyeonglee Did you consider adding some tests for this? I think it would be good to have some.
> 
> Good, I can. Do you mean under this directory? (https://github.com/ceph/ceph/tree/main/src/test/client)

That should be fine, or, under src/test/libcephfs. I'd prefer the former.

> And also, could you share a good example? (I think I cannot find original tests for sending commands thru libcephfs -> Client.)

Looks like there are none, which is embarrassing :/

You'd need to initiate a cephfs client and use the `[ceph_]mds_command` interface with `mds_spec=*` (obviously, you'd also need multiple active MDSs to test the result aggregation). While you are here, I'd request you to add a test for the single target.

Also, note that we have cephfs python bindings which also allows to send commands to the MDS

> def mds_command(self, mds_spec, args, input_data)

We also need to verify if this works as expected with this change. Again, I don't think we have tests for this. Sorry!

jinmyeonglee commented:
@vshankar OK, let me check what I understood.

1. let's add a single target and multi targets command test both to src/test/client -- and there are all c++ code-based tests. so maybe just Client class test only.
2. And also, add cephfs python binding test -- I agree, I already wrote some python scripts, so I can reuse this.

Bur for 2, where can I add python test script? under qa/somewhere?, Could you let me know?

vshankar commented:
> @vshankar OK, let me check what I understood.
> 
> 1. let's add a single target and multi targets command test both to src/test/client -- and there are all c++ code-based tests. so maybe just Client class test only.

Right.

> 2. And also, add cephfs python binding test -- I agree, I already wrote some python scripts, so I can reuse this.
> 
> Bur for 2, where can I add python test script? under qa/somewhere?, Could you let me know?

src/test/pybind/test_cephfs.py

jinmyeonglee commented:
@vshankar 
Hello, I have added test cases and fixed some subtle issues in the code. During the process, I considered that checking the JSON result of a specific command, like 'mds session ls', is not within our current scope. Instead, I focused on adding test cases to validate the results of both single target and multi-target commands. I would appreciate your feedback on these changes.

vshankar commented:
> @vshankar Hello, I have added test cases and fixed some subtle issues in the code. During the process, I considered that checking the JSON result of a specific command, like 'mds session ls', is not within our current scope. Instead, I focused on adding test cases to validate the results of both single target and multi-target commands. I would appreciate your feedback on these changes.

Thanks for letting me know - I'll have a look soon.

jinmyeonglee commented:
@vshankar Hello~! How long would this take?

vshankar commented:
> @vshankar Hello~! How long would this take?

Will be done today from cephfs side. I've additionally requested a review from @ljflores from core team.

ljflores reviewed (approved):
The core files here would be client and common. Conceptually, this seems fine. We should run it through the rados suite to make sure the client and common changes aren't breaking anything, but otherwise LGTM!

vshankar commented:
@yuriw - please include this in the next main branch integration run.

jinmyeonglee commented:
@vshankar Hello, I'm not trying to rush, but can you tell me when this QA will be finished?

vshankar commented:
> @vshankar Hello, I'm not trying to rush, but can you tell me when this QA will be finished?

@yuriw is running this through various test suites as you can see from his label. When the runs are done, Yuri tags (in trello) folks from various subsystem suites to review the runs. If everything looks fine, the change is merged.

I'll let @yuriw update about the run status.

rishabh-d-dave reviewed (changes requested):
Following test failed due to this PR - https://pulpito.ceph.com/rishabh-2023-09-12_12:12:15-fs-wip-rishabh-2023sep12-b2-testing-default-smithi/7394896

```
2023-09-13T08:49:21.956 INFO:tasks.workunit.client.0.smithi112.stderr:+ sudo python3 -m pytest -v /home/ubuntu/cephtest/clone.client.0/qa/workunits/fs/../../../src/test/pybind/test_cephfs.py
2023-09-13T08:49:22.277 INFO:tasks.workunit.client.0.smithi112.stdout:============================= test session starts ==============================
2023-09-13T08:49:22.277 INFO:tasks.workunit.client.0.smithi112.stdout:platform linux -- Python 3.8.10, pytest-4.6.9, py-1.8.1, pluggy-0.13.0 -- /usr/bin/python3
2023-09-13T08:49:22.278 INFO:tasks.workunit.client.0.smithi112.stdout:cachedir: .pytest_cache
2023-09-13T08:49:22.278 INFO:tasks.workunit.client.0.smithi112.stdout:rootdir: /home/ubuntu/cephtest/clone.client.0/src/test/pybind, inifile: pytest.ini
2023-09-13T08:49:22.436 INFO:tasks.workunit.client.0.smithi112.stdout:collecting ... collected 0 items / 1 errors
2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:
2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:==================================== ERRORS ====================================
2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:_______________________ ERROR collecting test_cephfs.py ________________________
2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:../../../clone.client.0/src/test/pybind/test_cephfs.py:914: in <module>
2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:    ???
2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:E   NameError: name 'with_setup' is not defined
2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:=========================== 1 error in 0.20 seconds ============================
```

joscollin commented:
> Following test failed due to this PR - https://pulpito.ceph.com/rishabh-2023-09-12_12:12:15-fs-wip-rishabh-2023sep12-b2-testing-default-smithi/7394896
> 
> ```
> 2023-09-13T08:49:21.956 INFO:tasks.workunit.client.0.smithi112.stderr:+ sudo python3 -m pytest -v /home/ubuntu/cephtest/clone.client.0/qa/workunits/fs/../../../src/test/pybind/test_cephfs.py
> 2023-09-13T08:49:22.277 INFO:tasks.workunit.client.0.smithi112.stdout:============================= test session starts ==============================
> 2023-09-13T08:49:22.277 INFO:tasks.workunit.client.0.smithi112.stdout:platform linux -- Python 3.8.10, pytest-4.6.9, py-1.8.1, pluggy-0.13.0 -- /usr/bin/python3
> 2023-09-13T08:49:22.278 INFO:tasks.workunit.client.0.smithi112.stdout:cachedir: .pytest_cache
> 2023-09-13T08:49:22.278 INFO:tasks.workunit.client.0.smithi112.stdout:rootdir: /home/ubuntu/cephtest/clone.client.0/src/test/pybind, inifile: pytest.ini
> 2023-09-13T08:49:22.436 INFO:tasks.workunit.client.0.smithi112.stdout:collecting ... collected 0 items / 1 errors
> 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:
> 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:==================================== ERRORS ====================================
> 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:_______________________ ERROR collecting test_cephfs.py ________________________
> 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:../../../clone.client.0/src/test/pybind/test_cephfs.py:914: in <module>
> 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:    ???
> 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:E   NameError: name 'with_setup' is not defined
> 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
> 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:=========================== 1 error in 0.20 seconds ============================
> ```

A rebase would fix this.

jinmyeonglee commented:
> > Following test failed due to this PR - https://pulpito.ceph.com/rishabh-2023-09-12_12:12:15-fs-wip-rishabh-2023sep12-b2-testing-default-smithi/7394896
> > ```
> > 2023-09-13T08:49:21.956 INFO:tasks.workunit.client.0.smithi112.stderr:+ sudo python3 -m pytest -v /home/ubuntu/cephtest/clone.client.0/qa/workunits/fs/../../../src/test/pybind/test_cephfs.py
> > 2023-09-13T08:49:22.277 INFO:tasks.workunit.client.0.smithi112.stdout:============================= test session starts ==============================
> > 2023-09-13T08:49:22.277 INFO:tasks.workunit.client.0.smithi112.stdout:platform linux -- Python 3.8.10, pytest-4.6.9, py-1.8.1, pluggy-0.13.0 -- /usr/bin/python3
> > 2023-09-13T08:49:22.278 INFO:tasks.workunit.client.0.smithi112.stdout:cachedir: .pytest_cache
> > 2023-09-13T08:49:22.278 INFO:tasks.workunit.client.0.smithi112.stdout:rootdir: /home/ubuntu/cephtest/clone.client.0/src/test/pybind, inifile: pytest.ini
> > 2023-09-13T08:49:22.436 INFO:tasks.workunit.client.0.smithi112.stdout:collecting ... collected 0 items / 1 errors
> > 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:
> > 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:==================================== ERRORS ====================================
> > 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:_______________________ ERROR collecting test_cephfs.py ________________________
> > 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:../../../clone.client.0/src/test/pybind/test_cephfs.py:914: in <module>
> > 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:    ???
> > 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:E   NameError: name 'with_setup' is not defined
> > 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
> > 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:=========================== 1 error in 0.20 seconds ============================
> > ```
> 
> A rebase would fix this.
> 
> EDIT: No, it should be dropped.

Hello, do you mean I need to rebase with master?

joscollin commented:
> > > Following test failed due to this PR - https://pulpito.ceph.com/rishabh-2023-09-12_12:12:15-fs-wip-rishabh-2023sep12-b2-testing-default-smithi/7394896
> > > ```
> > > 2023-09-13T08:49:21.956 INFO:tasks.workunit.client.0.smithi112.stderr:+ sudo python3 -m pytest -v /home/ubuntu/cephtest/clone.client.0/qa/workunits/fs/../../../src/test/pybind/test_cephfs.py
> > > 2023-09-13T08:49:22.277 INFO:tasks.workunit.client.0.smithi112.stdout:============================= test session starts ==============================
> > > 2023-09-13T08:49:22.277 INFO:tasks.workunit.client.0.smithi112.stdout:platform linux -- Python 3.8.10, pytest-4.6.9, py-1.8.1, pluggy-0.13.0 -- /usr/bin/python3
> > > 2023-09-13T08:49:22.278 INFO:tasks.workunit.client.0.smithi112.stdout:cachedir: .pytest_cache
> > > 2023-09-13T08:49:22.278 INFO:tasks.workunit.client.0.smithi112.stdout:rootdir: /home/ubuntu/cephtest/clone.client.0/src/test/pybind, inifile: pytest.ini
> > > 2023-09-13T08:49:22.436 INFO:tasks.workunit.client.0.smithi112.stdout:collecting ... collected 0 items / 1 errors
> > > 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:
> > > 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:==================================== ERRORS ====================================
> > > 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:_______________________ ERROR collecting test_cephfs.py ________________________
> > > 2023-09-13T08:49:22.471 INFO:tasks.workunit.client.0.smithi112.stdout:../../../clone.client.0/src/test/pybind/test_cephfs.py:914: in <module>
> > > 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:    ???
> > > 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:E   NameError: name 'with_setup' is not defined
> > > 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
> > > 2023-09-13T08:49:22.472 INFO:tasks.workunit.client.0.smithi112.stdout:=========================== 1 error in 0.20 seconds ============================
> > > ```
> > 
> > 
> > A rebase would fix this.
> > EDIT: No, it should be dropped.
> 
> Hello, do you mean I need to rebase with master?

I meant `with_setup` is dropped here 0bd813f069ed7a3ee691fbfb33da017bb00c1a55. So remove it from your code.

jinmyeonglee commented:
jenkins test make check

rishabh-d-dave reviewed (approved):
https://tracker.ceph.com/projects/cephfs/wiki/Main#22-Sep-2023

vshankar commented:
> https://tracker.ceph.com/projects/cephfs/wiki/Main#22-Sep-2023

@rishabh-d-dave has run this through fs suite and has approved on behalf of cephfs. This still needs to be run through rados suite. @yuriw isn't available for some more time I think, so, this has to wait till then or if someone from the core team (@ljflores?) plans to run it though the rados suite.

rishabh-d-dave commented:
jenkins test make check

rishabh-d-dave commented:
jenkins test make check amr64

rishabh-d-dave commented:
jenkins test make check arm64

pdvian commented:
https://pulpito.ceph.com/lflores-2023-09-06_18:19:12-rados-wip-yuri6-testing-2023-08-28-1308-distro-default-smithi/

Failures, unrelated:
1. https://tracker.ceph.com/issues/59142
2. https://tracker.ceph.com/issues/59196
3. https://tracker.ceph.com/issues/55347 
4. https://tracker.ceph.com/issues/62084
5. https://tracker.ceph.com/issues/62975                 <<<---- New Tracker
6. https://tracker.ceph.com/issues/62449
7. https://tracker.ceph.com/issues/61774
8. https://tracker.ceph.com/issues/53345

Details:
1. mgr/dashboard: fix e2e for dashboard v3 - Ceph - Mgr - Dashboard
2. ceph_test_lazy_omap_stats segfault while waiting for active+clean - Ceph - RADOS
3. SELinux Denials during cephadm/workunits/test_cephadm
4. task/test_nfs: AttributeError: 'TestNFS' object has no attribute 'run_ceph_cmd' - Ceph - CephFS
5. site-packages/paramiko/channel.py: OSError: Socket is closed - Infrastructure
6. test/cls_2pc_queue: TestCls2PCQueue.MultiProducer and TestCls2PCQueue.AsyncConsumer failure - Ceph - RGW
7. centos 9 testing reveals rocksdb "Leak_StillReachable" memory leak in mons - Ceph - RADOS
8. Test failure: test_daemon_restart (tasks.cephadm_cases.test_cli.TestCephadmCLI) - Ceph - Orchestrator

rishabh-d-dave commented:
> https://pulpito.ceph.com/lflores-2023-09-06_18:19:12-rados-wip-yuri6-testing-2023-08-28-1308-distro-default-smithi/
> 
> Failures, unrelated:
> 
>     1. https://tracker.ceph.com/issues/59142
> 
>     2. https://tracker.ceph.com/issues/59196
> 
>     3. https://tracker.ceph.com/issues/55347
> 
>     4. https://tracker.ceph.com/issues/62084
> 
>     5. https://tracker.ceph.com/issues/62975                 <<<---- New Tracker
> 
>     6. https://tracker.ceph.com/issues/62449
> 
>     7. https://tracker.ceph.com/issues/61774
> 
>     8. https://tracker.ceph.com/issues/53345
> 
> 
> Details:
> 
>     1. mgr/dashboard: fix e2e for dashboard v3 - Ceph - Mgr - Dashboard
> 
>     2. ceph_test_lazy_omap_stats segfault while waiting for active+clean - Ceph - RADOS
> 
>     3. SELinux Denials during cephadm/workunits/test_cephadm
> 
>     4. task/test_nfs: AttributeError: 'TestNFS' object has no attribute 'run_ceph_cmd' - Ceph - CephFS
> 
>     5. site-packages/paramiko/channel.py: OSError: Socket is closed - Infrastructure
> 
>     6. test/cls_2pc_queue: TestCls2PCQueue.MultiProducer and TestCls2PCQueue.AsyncConsumer failure - Ceph - RGW
> 
>     7. centos 9 testing reveals rocksdb "Leak_StillReachable" memory leak in mons - Ceph - RADOS
> 
>     8. Test failure: test_daemon_restart (tasks.cephadm_cases.test_cli.TestCephadmCLI) - Ceph - Orchestrator

@pdvian 
Meaning, RADOS tests were also successful, right?

I am asking because CephFS tests were also successful, if RADOS tests are also successful, we can merge this PR.

ljflores commented:
@rishabh-d-dave yes, rados is approved.
